exporters/zipkin/src/main/java/io/opentelemetry/exporters/zipkin/ZipkinSpanExporter.java

```diff
@@ -0,0 +1,192 @@
+/*
+ * Copyright 2020, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opentelemetry.exporters.zipkin;
+
+import static java.util.concurrent.TimeUnit.MICROSECONDS;
+import static java.util.concurrent.TimeUnit.NANOSECONDS;
+
+import io.opentelemetry.common.AttributeValue;
+import io.opentelemetry.common.AttributeValue.Type;
+import io.opentelemetry.sdk.trace.data.SpanData;
+import io.opentelemetry.sdk.trace.export.SpanExporter;
+import io.opentelemetry.trace.Span.Kind;
+import io.opentelemetry.trace.Status;
+import io.opentelemetry.trace.attributes.SemanticAttributes;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import javax.annotation.Nullable;
+import zipkin2.Endpoint;
+import zipkin2.Span;
+import zipkin2.codec.BytesEncoder;
+import zipkin2.reporter.Sender;
+
+/**
+ * This class was based on the OpenCensus zipkin exporter code at
+ * https://github.com/census-instrumentation/opencensus-java/tree/c960b19889de5e4a7b25f90919d28b066590d4f0/exporters/trace/zipkin
+ */
+final class ZipkinSpanExporter implements SpanExporter {
+
+  private static final Logger logger = Logger.getLogger(ZipkinSpanExporter.class.getName());
+
+  // The naming follows Zipkin convention. For http see here:
+  // https://github.com/openzipkin/brave/blob/eee993f998ae57b08644cc357a6d478827428710/instrumentation/http/src/main/java/brave/http/HttpTags.java
+  // For discussion about GRPC errors/tags, see here:  https://github.com/openzipkin/brave/pull/999
+  // Note: these 3 fields are non-private for testing
+  static final String GRPC_STATUS_CODE = "grpc.status_code";
+  static final String GRPC_STATUS_DESCRIPTION = "grpc.status_description";
+  static final String STATUS_ERROR = "error";
+
+  private static final Endpoint EMPTY_ENDPOINT = Endpoint.newBuilder().build();
+
+  private final BytesEncoder<Span> encoder;
+  private final Sender sender;
+
+  ZipkinSpanExporter(BytesEncoder<Span> encoder, Sender sender) {
+    this.encoder = encoder;
+    this.sender = sender;
+  }
+
+  static Span generateSpan(SpanData spanData) {
+    Endpoint endpoint = EMPTY_ENDPOINT;
```

Comment:
please don't add an empty endpoint

Reply:
The zipkin API just sets null when you pass in an empty one. Are you saying that if we can't make an endpoint, we shouldn't send spans?

Reply:
it is distracting to set an empty endpoint for reasons including what you mentioned. that's the main point. just call the endpoint builder when you have one?

Reply:
the longer answer is that it would be easy for this code to drift to adding more data into the span for no reason. For example, if you change codec later, it would likely serialize `"localEndpoint": {}` which is useless overhead, plus adding empty is already confusing.

hope this makes sense.